### ISSUE TITLE
Parse Dates from the French Revolutionary calendar and Roman Numbers

### ISSUE THREAD

**Issue by boerni667** [scrapinghub/dateparser#848]
It would be nice if the dateparser would support Dates based on the french Revolutionary Calender and Dates written in Roman numbers.

**Comment by noviluni** [comment]
Hi @boerni667!

We started the [`number-parser` project](https://github.com/scrapinghub/number-parser) (it is at an early stage) to parse numbers with the idea of integrating it with `dateparser` once it's more stable. We will probably add support for roman numbers there and we could use it here.

Thanks for the idea! I will take it into consideration.

**Comment by boerni667** [comment]
Yeah, i've got test data set containing redicoulous dates and try to parse that into machine-readable dates.


e.g.:

zgrep ^An testdata/260c264c.txt.gz

An 10 [1802]
An 1 (1792/93]
An 1 [1792/93]
An 1. [1792/93]
An 12 1804
An 13 [1805]
An. 1653
An. 1737
An. 2 [1793/1794]
An 2. [1793/94]
An 2. de la République française, une et indivisible [i.e. 1793/94]
An 3 [1794]
An 3 [1794/95]
An 3.eme [=1795]
An 3.me [=1795]
An 3.me de la République
An 4 [i.e. 1795/96]
An 5 [1796/97]
An 5 - 1797
An 5 [1797]
An 5e
An 5e.
An 6 [1797/98]
An 6 [1798]
An 7 [1798/99]
An 7 [1799]
An 7 [i.e. 1798/99]
An 8 [1800]
An 9(1801)
An 9 [i.e. 1801]
Andreas Lorensen
Anfangs
Angefangen Im der nunmehro letzten itziger Weltt Zeitt Vnd Jahre Nach Jesu Chr[ist]i Vnsers seligmachers gebuhrtt 1611
Angefangen im Jahr 1763 und fortgefahren bis 1798
Angefangen in Görlitz 1800. den 10. Januar und völlig zu Ende gebracht 1800 den 31. Decbr:
An I [1792]
An II [1793]
An II [1793/94]
An II [1794]
An III
An III [1794]
An III, 1794/95
An III. [1794/95]
An III [=1795]
An III [1795]
An III [i.e. 1794/1795]
An III [i. e. 1795]
An III [i.e. 1795]
An IV [1795/1796]
An IV [1795/96]
An IVme. de la Républ. Franç. [1796]
An IX - 1800
An IX [1800]
An IX [1800/01]
An IX (1801)
An IX [1801]
An IX 1801
An IX [i.e. 1801]
Annex 1 - 7
Anno. 1499
Anno 1588.
Anno 1620.
Anno 1628.
An[n]o 1648
Anno 1652
Anno 1667
Anno: 1677. d. 2. Octobr.
Anno 1678
Anno 1681
Anno 1682.
Anno 1685
Anno 1686
Anno 1686 die 27 August[us]
Anno 1691
Anno 1709
Anno 1720.
Anno 1729
Anno 1737.
Anno 1749
Anno 1751
Anno [1829]
Anno 31 1631
Anno Christi Hiesu Millesimoquingentesimotertio
Anno Christi MDCCXVII.
Anno dn̄i 1501 nona die mēsi Maij
Anno domini. 1.4.9.9. die ... Julij. 27 [27. Juli 1499]
Anno domini. M.ccccc. die. xv. mensis Iulij [15. Juli 1500]
Anno d[omi]ni M.cccc.lxxi xi k[a]l[endas] decembris [21. November 1471]
Anno domini MCCCCLXXVIIII. Die XX. Decembris
Anno domini. M.cccc.lxxviij
Anno d[omi]ni Mcccclxxxix. xij. kal[endis] me[n]s[is] Aprilis [21. März 1489]
Anno d[omi]ni. M.cccc.lxxxvij.
Anno d[omi]ni. M.cccc.xc. Decimasexta die mensis Maij
Anno d[omi]ni M.cccc.xci
Anno domini. M.cccc.xciiij [26. Mai 1494]
Anno domini M.cccc.xcij. Decima Februarij [10. Februar 1492]
Anno d[omi]ni M.cccc.xcv.
Anno domini. M.cccc.xcv.
Anno domini. M.ccccxcvj.
Anno Domini M.D.XL.
Anno Domini M.D.XXXIX.
Anno domini Millesimoquadragintesimo nonagesimoseptimo. q[ua]rto Calendas Maij. [15. und 28. April 1497]
Anno domini Millesimoquadringentesimo Nonagesimoquinto
Anno domini Millesimoquadringentesimooctuagesimo die vero tricesimoprimo mensis May
Anno d[omi]ni millesimoquadringentesimo septuagesimo p[ri]mo Octaua die nouembris [8. November 1471]
Anno d[omi]ni Millesimo q[ui]ngentesimo
Anno finiente Judicium [1613]
Anno gratie. M.cccc.xciiij.
Anno gratie. M.cccc.xcviij.
Anno incarnate deitatis. M.cccc.lxxxiiij. die vltima Julij [31. Juli 1484]
Anno incarnationis domini post millesimum quaterq[ue] centesimum octogesimooctauo. Decimo vero kalen[das]. mensis Octobris [22. September 1488]
Anno jubileo millesimo q[ui]nge[n]tesimo. decima die martij [10. März 1500]
Anno. M.ccccc.xvi. Kal. Martij.
Anno MDCCIV
Anno M.DCC.LIV.
Anno MDCCX.
Anno MDCCXLVIIII.
Anno MDCCXXX.
Anno MDCLII.
Anno M DC LIII
Anno MDCLXVIII
Anno MDCLXVIII. ... XXVII. Martii
Anno M.DC.LXXIX. d. 1. Maji
Anno MDCLXXVII
Anno MDCLXXVII.
Anno MDCLXXXIIX
Anno MDCLXXXIX. M. Jan.
Anno M.DC.LXXXVI
Anno MDCLXXXVI
Anno MDCLXXXVII
Anno M. DC. XIII.
Anno M.DC.XLI.
Anno M. DC. XVI.
Anno M.DC.XVI.
Anno M.DC.XVII.
Anno M. DC. XXI.
Anno M. DC. XXII
Anno MDCXXVIII.
Anno M.D.LXIX.
Anno M. D. LXXX.
Anno M. D. LXXXVI.
Anno M.D.XC.
Anno M.D.XXIII. mense Iulio
Anno M.D.XXXII.
Anno nostre salutis. [14]97
Anno nostre salutis Millesimo q[ua]dringentesimo nonagesimonono
Anno O. R. M.DCXC. A. D. XXII Maji
Anno O. R. S. 1680 Die XXIV. Julii
Anno saluatoris nostri. 1496 Decimono kalendas septembris
Anno salutifere incarnationis Millesimo quadringentesimo nonagesimotertio
Anno salutis. 1495
Anno salutis christiane. Millesimo. quatercentesimo. Nonagesimo septimo. Idibus Maii. 1.4.9.7. [15. Mai 1497]
Anno salutis. M.CCCC.LXXXIII [17. Mai 1483]
Anno salutis. M.cccc.lxxxix xviij. k[a]l[endas] Nouembris [15.X.1489]
Anno salutis. M.cccc.xcv.
Anno VI [1797/98]
Anno VI. [1797/98]
Anno virginei partus Millesimoq[ua]dringentesimononagesimoqui[n]to
Anno XI (1803)
An Quatrième [1796]
An Sept [1798/1799]
An Sept. [1798/1799]
An V.
An V [1796/97]
An V [1797]
An V. - 1797
An VI
An VI [1797]
An VI [1797/1798]
An VI [1798]
An. VI. de la République [1798]
An VII$e1798
An VII [1798/1799]
An VII [1798/99]
An VII. 1799
An VII de la République
An VI [i.e. 1794]
An VI [i.e. 1798]
An VIII [1799]
An VIII [1799/1800]
An VIII [1800]
An VII [i.e. 1799]
An VII [i.e.1799]
An VIII [i.e. 1799/1800]
An VIII [i.e. 1800]
An VIII [i.e.1800]
An X - 1801
An X [1801]
An X [1801/02]
An X - 1802
An X = 1802
An X [1802]
An X. - 1802
An X. = 1802
An. X [1802]
An X De La République [1802]
An XI [1801]
An XI. (1802)
An XI [1802/03]
An XI [1802/1803]
An XI (1803)
An XI - 1803
An XI [1803]
An XI. (1803)
An XI. [1803]
An X [i.e. 1802]
An XII [1802]
An XII [1803/04]
An XII = 1804
An XII [1804]
An XII [1805]
An XI. [i.e. 1802]
An XI [i.e.1803]
An XIII.
An XIII (1805)
An XIII - 1805
An XIII. - 1805
An XIV - 1805
An XIV. - 1805
An XIV [1806]
An XIV [1807]
